configure.ac

```diff
@@ -209,8 +209,23 @@ PKG_CHECK_MODULES(zlib,
                  [CXXFLAGS="$zlib_CFLAGS $CXXFLAGS"
                  LIBS="$zlib_LIBS $LIBS"])
 
+# Check if compiler's default C++ mode is >= C++11
```

Comment:
This looks unnecessary complex to me, so I tried myself:
https://github.com/qbittorrent/qBittorrent/commit/fe25b4f72a4b86478af7c07ecb428e731b828608
https://travis-ci.org/Chocobo1/qBittorrent/builds/510507927
Seems it get the job done too.

Reply:
Here's my reasoning: (it might be flawed)
1. At this point CXXFLAGS are populated from the searched packages via pkg-config. Some package might have inserted a `-std=` flag. That's why I temporarily empty them, to truely test the compiler's default mode.
2. If I understand it correctly, qmake cares if at least c++11 is default. If it is, then it doesn't put a `-std=gnu++11` flag. So I am using the same condition/test to force c++14 via `CONFIG`. If the default is >c++11 then we investigate if it satisfies at least c++14 by itself. This, theoritically, will work for a future gcc release where the default will be c++17. In that case, we don't want to hardcode c++14 for qbittorrent, because most likely the system's libtorrent will have been built in c++17.

PS: After seeing your diff: In the case where we need to explicitly force c++14 maybe it makes more sense to just `CONFIG += c++14` instead of `-std=c++14`.   <--- All this in addition to my diff IMO. In that case, my diff needs to be reworked to skip some tests.

Reply:
>That's why I temporarily empty them, to truely test the compiler's default mode.

Although I'm not sure what is the case in reality, it is correct to be cautious.

>In that case, we don't want to hardcode c++14 for qbittorrent, because most likely the system's libtorrent will have been built in c++17.

I believe our current implemention in master branch already do this.

>my diff needs to be reworked to skip some tests.

yes, especially the test about c++11.
I updated my commit: https://github.com/Chocobo1/qBittorrent/commit/e5d3f03d9046f22f0434988ecd4bfeae19719685
https://travis-ci.org/Chocobo1/qBittorrent/builds/510816027
Perhaps I should push to this PR for review?

Reply:
>Perhaps I should push to this PR for review?

PR updated.